ons-spark/spark-concepts/persistance.md

```diff
@@ -0,0 +1,89 @@
+
+### Filling up cache/file system with persisted data
+
+There is a limited amount of memory on the executors of the Spark cluster and if you fill it up with cached data Spark will start to spill data onto disk or return an out of memory error. Spark will prioritise recent caches and run more expecsive processes to manage the low priority data. Remember to empty the cache when you've stopped using a DataFrame and don't cache too many DataFrames at one time. Memory management in a Spark cluster is a complex topic, a good introduction is given in [this blog](https://0x0fff.com/spark-memory-management).
+
+The same is true for the file system, although there is much more space to store data of course. It's good practice to delete a checkpoint directory after use within a Spark script. The same is true for staging tables, unless you want to view the data at a later date.
+
+
+## Use cases for persisting
+
+- It usually makes sense to cache data in memory before repeated actions are applied to the DataFrame, like the example in the **Caching** article. 
+- Another use case for caching in memory is just before running an iterative algorithm, where you make use of the cached DataFrame in each iteration.
+- Also, when running a model on a DataFrame multiple times whilst trying out different inputs or parameters. 
+- As shown in the **Checkpoint** article, when a DataFrame's lineage gets long or complex breaking up the lineage by persisting can improve performance.
+- Caching is also a great tool whilst developing your code. For example, perhaps you have just linked two data sets and want to do some checks to make sure the result makes sense. Perhaps the checks would involve running multiple actions e.g. `.show()`, `.count()`, `.describe()`, but without caching after the join Spark would repeat the join process for each action. Once you're happy with the result you might want to empty the cache and possibly even replace the cache with a staging table so that users can quality assure the result of the join after each run of the pipeline. 
+
+### Checkpoint or staging tables?
+
+**Checkpoint**
+
+- Checkpointing is more convenient than staging tables in terms of writing the code
+- Overall it can be more effort to maintain staging tables, especially if using the option to insert into a table, as you will have to alter the table if the DataFrame structure changes
+
+**Staging tables**
+
+- The same table can be overwritten, meaning there is no need to clean up old checkpointed directories
+- It is stored in a location that is easier to access, rather than the checkpointing folder, which can help with debugging and testing changes to the codeThey can be re-used elsewhere
+- If .insertInto() is used, you can take advantage of the table schema, as an exception will be raised if the DataFrame and table schemas do not match
```

Comment:
Code formatting needed for insertInto(), although I'll check all these when adding links anyway